code/__HELPERS/lists.dm

```diff
@@ -367,10 +372,15 @@
 			i++
 	return i
 
-/proc/count_by_name(var/list/atom/L, name)
+/proc/count_by_name(var/list/L, name)
 	var/i = 0
-	for(var/atom/T in L)
-		if(T.name == name)
+	var/atom/current_atom = null
+	var/current_name = ""
+	for(var/T in L)
+		if(isatom(T) || ispath(T))
+			current_atom = T
+		current_name = ispath(T) ? initial(current_atom.name) : isatom(T) ? current_atom.name : T
```

Comment:
wouldn't this still runtime when passed an item path?

Reply:
Didn't when I tested it